project/Doc.scala

```diff
@@ -91,3 +91,57 @@ object ScaladocNoVerificationOfDiagrams extends AutoPlugin {
     Scaladoc.validateDiagrams in Compile := false
   )
 }
+
+/**
+ * Unidoc settings for root project. Adds unidoc command.
+ */
+object UnidocRoot extends AutoPlugin {
+
+  object autoImport {
+    val unidocProjectExcludes = settingKey[Seq[ProjectReference]]("Excluded unidoc projects")
+  }
+  import autoImport._
+
+  object CliOptions {
+    val genjavadocEnabled = CliOption("akka.genjavadoc.enabled", false)
+  }
+
+  override def trigger = noTrigger
+
+  val akkaSettings = UnidocRoot.CliOptions.genjavadocEnabled.ifTrue(Seq(
+    javacOptions in (JavaUnidoc, unidoc) ++= Seq("-Xdoclint:none"),
+    // genjavadoc needs to generate synthetic methods since the java code uses them
+    scalacOptions += "-P:genjavadoc:suppressSynthetic=false",
+    // FIXME: see #18056
+    sources in(JavaUnidoc, unidoc) ~= (_.filterNot(_.getPath.contains("Access$minusControl$minusAllow$minusOrigin")))
+  )).getOrElse(Nil)
+
+  val settings = inTask(unidoc)(Seq(
+    unidocProjectFilter in ScalaUnidoc := inAnyProject -- inProjects(unidocProjectExcludes.value: _*),
+    unidocProjectFilter in JavaUnidoc := inAnyProject -- inProjects(unidocProjectExcludes.value: _*),
+    apiMappings in ScalaUnidoc := (apiMappings in (Compile, doc)).value
```

Comment:
I know that this is the old code, but this setting might not be needed anymore after https://github.com/sbt/sbt-unidoc/pull/10

But to be sure, we need to test, if scaladoc generated by `unidoc` has apiMappings to external libs.